packages/subiquity_client/lib/src/server.dart

```diff
@@ -11,6 +11,25 @@ enum ServerMode { LIVE, DRY_RUN }
 /// @internal
 final log = Logger('subiquity_server');
 
+/// Vocabulary type that combines an [InternetAddress] and a [port] number
+class Endpoint {
+  final InternetAddress address;
+  final int port;
+
+  /// Creates an Unix domain socket endpoint.
+  factory Endpoint.unix(String socketPath) => Endpoint(
+        InternetAddress(socketPath, type: InternetAddressType.unix),
+        port: 0,
+      );
+
+  /// Creates a localhost TCP endpoint on port number [port].
+  factory Endpoint.localhost(int port) => Endpoint(
+        InternetAddress("127.0.0.1", type: InternetAddressType.IPv4),
```

Comment:
i thought we had prefer_single_quotes enabled :D

Reply:
That deserves a PR on its own :D

![image](https://user-images.githubusercontent.com/11138291/171682707-c5557715-e6ad-41ff-8fe8-c39eb1b22bfc.png)